libraries/AP_Button/AP_Button.cpp

```diff
@@ -186,9 +326,16 @@ void AP_Button::send_report(void)
 void AP_Button::setup_pins(void)
 {
     for (uint8_t i=0; i<AP_BUTTON_NUM_PINS; i++) {
+        if (is_pwm_input(i)) {
+            // TESTME: what happens if this is set to an invalid value?  Error
+            // message at high rate?!
```

Comment:
needs testing

Reply:
Tested.  We protect against this the pwm-pin-source class.